app/models/marc_field.rb

```diff
@@ -0,0 +1,138 @@
+###
+#  MarcField is the base class to provide common
+#  behavior for other classes to inherit from
+###
+class MarcField
+  attr_reader :document, :tags
+  delegate :present?, to: :relevant_fields
+
+  def initialize(solr_document, tags = [])
+    @document = solr_document
+    @tags = tags
+    run_preprocessors
+  end
+
+  def inspect
+    "#<#{self.class.name}:0x#{object_id} @document=#<SolrDocument:#{document.id}>>"
+  end
+
+  def label
+    I18n.t(
+      "searchworks.marc_fields.#{i18n_key}.label",
+      default: :"searchworks.marc_fields.default.label"
+    )
+  end
+
+  def values
+    relevant_fields.map do |field|
+      field.subfields.map(&:value).join(subfield_delimeter)
+    end
+  end
+
+  def to_partial_path
+    'marc_fields/marc_field'
+  end
+
+  private
+
+  def marc
+    return {} unless document.respond_to?(:to_marc)
+    document.to_marc
+  end
+
+  def relevant_fields
+    return [] unless marc.present?
+    @relevant_fields ||= marc.fields(tags)
+  end
+
+  def remove_hidden_indicators
+    relevant_fields.reject! do |field|
+      (Constants::HIDE_1ST_IND.include?(field.tag) && field.indicator1 == '1') ||
+        (Constants::HIDE_1ST_IND0.include?(field.tag) && field.indicator1 == '0')
+    end
+  end
+
+  def remove_hidden_subfields
+    relevant_fields.each do |field|
+      field.subfields.reject! do |subfield|
+        Constants::EXCLUDE_FIELDS.include?(subfield.code)
+      end
+    end
+  end
+
+  def merge_matched_vernacular_fields
+    relevant_fields.each_with_index do |field, index|
+      next if field.tag =~ /^880/
+      next unless field_has_vernacular_matcher?(field)
+      vernacular_field = matching_vernacular_field(field)
+      relevant_fields.insert(index + 1, vernacular_field) if vernacular_field
+    end
+  end
+
+  def append_unmatched_vernacular_fields
+    marc.fields(['880']).each do |vernacular_field|
+      next unless field_has_vernacular_matcher?(vernacular_field)
+      tag, matcher = vernacular_field_tag(vernacular_field)
+      next unless tags.include?(tag) && matcher == '00'
+      relevant_fields.insert(tag_indexes[tag], vernacular_field)
+    end
+  end
+
+  def tag_indexes
```

Comment:
skimming this class, everything makes sense to me except this method.  Perhaps a method comment would be useful?